### ISSUE TITLE
Training Epochs and Overfitting

### ISSUE THREAD

**Issue by danabez1** [UKPLab/sentence-transformers#1194]
Hello, 

I am curious to understand why only 1 epoch was used to train the original sentence-BERT? 
Have you tested it with more than 1 epoch? What was the result then? 

I tried to follow your code to fine-train SBERT using a pre-trained model ('all-mpnet-base-v2') with SoftmaxLoss on my data (10K samples) and got overfitting after 1 epoch. 
I used SoftmaxLoss (similar to the original paper).

What kind of methods do you recommend to handle overfitting?
In general, should I use one of the pre-trained models for fine-tuning or is it misused?
I will be glad for any recommendations on fine-tuning SBERT.

Thanks,
Dana

**Comment by nreimers** [comment]
SoftmaxLoss is a horrible loss and should be avoided. There is no compelling explanation why SoftmaxLoss works.

Use MultipleNegativesRankingLoss instead, way more intuitive and much better results.

**Comment by danabez1** [comment]
In addition,  how much data is recommended to use for fine tuning SBERT?

**Comment by nreimers** [comment]
Not sure how the behavior is with SoftmaxLoss.

With NegativeRankingLoss: More data is better. The most recent models are trained on over 1 billion training pairs.

**Comment by danabez1** [comment]
Thanks. I thought of using TripletLoss, as I cannot ensure that all data points are from different types - Is that ok?

Still, I wonder about the original sentence BERT model - why only 1 epoch was used to train the original sentence-BERT?
Have you tested it with more than 1 epoch? What was the result then?

In addition, what kind of method do you recommend to use in case of overfitting?

**Comment by danabez1** [comment]
Hi, another question -
I would like to train sentence-bert on a specific domain to be used for sencene similarity task.  
Do you recommend when training the model to start from distilroberta-base model (or any other base model) or from a pre-trained model, like paraphrase-MiniLM-L12-v2.
Thanks :)
